### PR TITLE
fix alert notify template lack info

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
pod and workload alert lack info

Solution:
update alert notify template

Issue:
https://github.com/rancher/rancher/issues/18156
https://github.com/rancher/rancher/issues/18160

### CODE REVIEW

File: pkg/controllers/user/alert/common/common.go [hunk 1]
```diff
@@ -19,3 +19,7 @@ func GetAlertManagerSecretName(appName string) string {
 func GetAlertManagerDaemonsetName(appName string) string {
 	return fmt.Sprintf("alertmanager-%s", appName)
 }
+
+func GetProjectAlertDisplayName(projectDisplayName, projectID string) string {
```
Comment (orangedeng) [inline pkg/controllers/user/alert/common/common.go]: I think cluster id should have display name too.

Comment (aiwantaozi) [reply pkg/controllers/user/alert/common/common.go]: updated

File: pkg/controllers/user/alert/configsyncer/configsyncer.go [hunk 2]
```diff
@@ -86,6 +92,12 @@ func (d *ConfigSyncer) sync() error {
 		return nil
 	}
 
+	cluster, err := d.clusterLister.Get(metav1.NamespaceAll, d.clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "Get cluster %s failed", d.clusterName)
+	}
+	d.clusterDisplayName = common.GetClusterAlertDisplayName(cluster.Spec.DisplayName, d.clusterName)
```
Comment (orangedeng) [inline pkg/controllers/user/alert/configsyncer/configsyncer.go]: Multiple go routine will change the display name in the same time

**Comment by orangedeng** [comment]
LGTM

File: pkg/controllers/user/alert/watcher/pod.go [hunk 3]
```diff
@@ -172,17 +173,23 @@ func (w *PodWatcher) checkPodRestarts(pod *corev1.Pod, alert *v3.ProjectAlertRul
 				if err != nil {
 					logrus.Warnf("Failed to get cluster for %s: %v", w.clusterName, err)
 				} else {
-					clusterDisplayName = cluster.Spec.DisplayName
+					clusterDisplayName = common.GetClusterAlertDisplayName(cluster.Spec.DisplayName, w.clusterName)
```
Comment (alena1108) [inline pkg/controllers/user/alert/watcher/pod.go]: I feel like code on line 176-183 repeats in 277-284. We should extract it to the common method, and use it

Comment (aiwantaozi) [reply pkg/controllers/user/alert/watcher/pod.go]: updated

**Comment by aiwantaozi** [comment]
@alena1108 The pr is updated, could you check?

**Comment by alena1108** [comment]
LGTM

**Pull request merged**